perfkitbenchmarker/pkb.py

```diff
@@ -29,25 +29,29 @@
          will then return the machine to the state it was at before Prepare
          was called.
 
-PerfKitBenchmarker has following run stages: prepare, run, cleanup and all.
-
-prepare: PerfKitBenchmarker will read command-line flags, decide which
-         benchmarks to run and create necessary resources for each benchmark,
-         including networks, VMs, disks, keys and execute the Prepare function
-         of each benchmark to install necessary software, upload datafiles, etc
-         and generate a run_uri, which can be used to run benchmark multiple
-         times.
+PerfKitBenchmarker has the following run stages: provision, prepare,
+    run, cleanup, teardown, and all.
+
+provision: Read command-line flags, decide what benchmarks to run, and
+    create the necessary resources for each benchmark, including
+    networks, VMs, disks, and keys, and generate a run_uri, which can
+    be used to run the benchmark multiple times.
+prepare: Execute the Prepare function of each benchmark to install
+         necessary software, upload datafiles, etc.
 run: PerfKitBenchmarker execute the Run function of each benchmark and collect
      samples generated. Publisher may publish these samples accourding to
      settings.  Run stage can be called multiple times with the run_uri
-     generated by prepare stage.
+     generated by provision stage.
 cleanup: PerfKitBenchmarker will run Cleanup function of each benchmark to
-         uninstall software and delete data files. Then it will delete VMs, key
-         files, networks, disks generated in prepare stage.
-all: PerfKitBenchmarker will run all above stages (prepare, run, cleanup). Any
-     resources generated in prepare will be automatically deleted at last.
-     PerfKitBenchmarker won't be able to rerun with exactly same VMs, networks,
-     disks with the same run_uri.
+         uninstall software and delete data files.
+teardown: Delete VMs, key files, networks, and disks created in the
+    'provision' stage.
+
+all: PerfKitBenchmarker will run all of the above stages (provision,
+     prepare, run, cleanup, teardown). Any resources generated in the
+     provision stage will be automatically deleted in the teardown
```

Comment:
It may be worth noting that they'll be deleted even if there's an error during any of the stages.